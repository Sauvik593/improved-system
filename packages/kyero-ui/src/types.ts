import type { LinkHTMLAttributes } from 'react';

export type BaseLinkProps =
  | {
      to: string;
    }
  | {
      to: {
        search?: string;
        pathname?: string;
      };
    };

export type LinkProps = BaseLinkProps & LinkHTMLAttributes<HTMLAnchorElement>;

export type LinkComponentProps = {
  className: string;
  children: React.ReactNode;
} & LinkProps;
