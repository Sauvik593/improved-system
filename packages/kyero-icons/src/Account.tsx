export const Account = () => {
  return (
    <svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
      <path
        d="M5.625 8.125C5.625 9.28532 6.08594 10.3981 6.90641 11.2186C7.72688 12.0391 8.83968 12.5 10 12.5C11.1603 12.5 12.2731 12.0391 13.0936 11.2186C13.9141 10.3981 14.375 9.28532 14.375 8.125C14.375 6.96468 13.9141 5.85188 13.0936 5.03141C12.2731 4.21094 11.1603 3.75 10 3.75C8.83968 3.75 7.72688 4.21094 6.90641 5.03141C6.08594 5.85188 5.625 6.96468 5.625 8.125V8.125Z"
        stroke="#1F4DEF"
        strokeWidth="1.5"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
      <path
        d="M15.7609 17.3968C15.0061 16.6372 14.1085 16.0344 13.1199 15.6231C12.1312 15.2118 11.0709 15 10.0001 15C8.92926 15 7.869 15.2118 6.88032 15.6231C5.89165 16.0344 4.99408 16.6372 4.23926 17.3968"
        stroke="#1F4DEF"
        strokeWidth="1.5"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
      <path
        d="M0.625 10C0.625 12.4864 1.61272 14.871 3.37087 16.6291C5.12903 18.3873 7.5136 19.375 10 19.375C12.4864 19.375 14.871 18.3873 16.6291 16.6291C18.3873 14.871 19.375 12.4864 19.375 10C19.375 7.5136 18.3873 5.12903 16.6291 3.37087C14.871 1.61272 12.4864 0.625 10 0.625C7.5136 0.625 5.12903 1.61272 3.37087 3.37087C1.61272 5.12903 0.625 7.5136 0.625 10V10Z"
        stroke="#1F4DEF"
        strokeWidth="1.5"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </svg>
  );
};
