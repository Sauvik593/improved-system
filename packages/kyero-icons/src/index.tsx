export * from './AnimatedSpinner';
export * from './ArrowUp';
export * from './Account';
export * from './ArrowDown';
export * from './Advertise';
export * from './ArrowBackward';
export * from './ArrowForward';
export * from './AlertInfo';
export * from './Avatar';
export * from './BoostRocket';
export * from './Camera';
export * from './Cart';
export * from './Close';
export * from './Chat';
export * from './Checked';
export * from './ChevronDown';
export * from './ChevronLeft';
export * from './Dashboard';
export * from './DeleteIcon';
export * from './Edit';
export * from './Email';
export * from './Eye';
export * from './Floorplans';
export * from './Forbidden';
export * from './Furniture';
export * from './Gate';
export * from './Globe';
export * from './Golf';
export * from './Hamburger';
export * from './HouseIcon';
export * from './JoinLogo';
export * from './List';
export * from './Logout';
export * from './LogoWide';
export * from './Imports';
export * from './InfoIcon';
export * from './Kyero';
export * from './NextIcon';
export * from './Outside';
export * from './Parking';
export * from './Photo';
export * from './Plus';
export * from './Pool';
export * from './PrevIcon';
export * from './Processing';
export * from './Properties';
export * from './Quote';
export * from './RemoteViewing';
export * from './Reply';
export * from './School';
export * from './Search';
export * from './ShareIcon';
export * from './Star';
export * from './SunMark';
export * from './Temperature';
export * from './Tick';
export * from './Town';
export * from './QuestionMark';
export * from './WavyArrow';
export * from './Views';
export * from './VirtualTour';
export * from './XIcon';
