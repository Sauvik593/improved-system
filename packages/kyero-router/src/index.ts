import { DocsRoutes } from './docs-routes';
import { GuidesRoutes } from './guides-routes';
import { IDRoutes } from './id-routes';
import { JoinRoutes } from './join-routes';
import { SupportRoutes } from './support-routes';
import { DataRoutes } from './data-routes';
import { returnCountryRouteForLocale } from './helpers';

import { preloadRoutesFromAPI } from './preload-routes-from-api';

import type {
  RoutesFromFrontendAPI,
  Countries,
  Locales,
  CountryPreloadedRoute,
  ExternalRoutes,
  FrontendRoutes,
  SocialRoutes,
} from './types';

export const Routes = {
  id: IDRoutes,
  join: JoinRoutes,
  guides: GuidesRoutes,
  docs: DocsRoutes,
  data: DataRoutes,
  support: SupportRoutes,
};

export class KyeroRouter {
  preloadedRoutes: RoutesFromFrontendAPI | null;

  constructor() {
    this.preloadedRoutes = null;
  }

  async preloadRoutes() {
    this.preloadedRoutes = await preloadRoutesFromAPI();
  }

  getExternalRoutes(locale: string) {
    return {
      privacy: Routes.docs.privacyURL(locale),
      terms: Routes.docs.termsURL(locale),
      cookies: Routes.docs.cookiesURL(locale),
      buyersGuide: Routes.guides.buyersGuideURL(locale),
      spainGuide: Routes.guides.spainGuideURL(locale),
      portugalGuide: Routes.guides.portugalGuideURL(locale),
      franceGuide: Routes.guides.franceGuideURL(locale),
      italyGuide: Routes.guides.italyGuideURL(locale),
      pdfGuide: Routes.guides.pdfGuideURL(locale),
      podcasts: Routes.guides.podcastsURL(),
      support: Routes.support.url(),
      agentsFAQ: Routes.support.agentFAQURL(locale),
      generalFAQ: Routes.support.generalFAQURL(locale),
      buyersFAQ: Routes.support.buyersFAQURL(locale),
      join: Routes.join.joinURL(locale),
      joinContact: Routes.join.contactURL(locale),
      marketData: Routes.data.marketData(locale),
      affordability: Routes.data.affordabilityCalculator(locale),
      jobs: 'https://jobs.kyero.com',
      account: Routes.id.settingsURL(locale),
      forgotPassword: Routes.id.passwordResetURL(locale),
    };
  }

  getSocialRoutes() {
    return {
      facebook: 'https://www.facebook.com/kyero.co.uk/',
      twitter: 'https://twitter.com/kyero',
      instagram: 'https://www.instagram.com/kyero_property/',
      linkedin: 'https://www.linkedin.com/company/kyero-com/',
      youtube: 'https://www.youtube.com/channel/UCmgKXDJ0fvVzMAAp4YlMYxA',
      pinterest: 'https://www.pinterest.co.uk/kyeroproperty/',
    };
  }

  getFrontendRoutes(locale: string) {
    const { preloadedRoutes } = this;

    if (!preloadedRoutes) {
      throw new Error('Routes not preloaded');
    }

    return {
      properties_for_sale: returnCountryRouteForLocale(preloadedRoutes.properties_for_sale, locale),
      agents_search: returnCountryRouteForLocale(preloadedRoutes.agents_search, locale),
      properties_to_rent: returnCountryRouteForLocale(preloadedRoutes.properties_to_rent, locale),
      cookie_preferences: preloadedRoutes.cookie_preferences[locale as Locales],
      sitemap: preloadedRoutes.sitemap[locale as Locales],
    };
  }
}

export type {
  Countries,
  Locales,
  CountryPreloadedRoute,
  ExternalRoutes,
  FrontendRoutes,
  SocialRoutes,
};
