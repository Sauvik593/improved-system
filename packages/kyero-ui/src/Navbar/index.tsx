export * from './Navbar';
export * from './NavElement';
