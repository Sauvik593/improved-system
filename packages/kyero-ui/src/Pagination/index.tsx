import cn from 'classnames';
import { Page } from './Page';
import { Button } from './Button';
import { usePagination } from './usePagination';

import type { LinkComponentProps } from '../types';

export interface PaginationProps {
  totalPages: number;
  currentPage: number;
  pathname: string;
  nextPage: number | null;
  prevPage: number | null;
  LinkComponent?: (props: LinkComponentProps) => React.ReactElement;
  search: string;
}

const DefaultComponent = ({ to, children, className }: LinkComponentProps) => {
  const path = typeof to === 'string' ? to : `${to?.pathname}?${to?.search}`;
  return (
    <a href={path} className={cn(className)}>
      {children}
    </a>
  );
};

export const Pagination = ({
  currentPage,
  totalPages,
  pathname,
  prevPage,
  nextPage,
  LinkComponent = DefaultComponent,
  search,
}: PaginationProps) => {
  const { showPrevDots, showNextDots, pagesArray } = usePagination(search, currentPage, totalPages);
  return (
    <nav className="mt-4">
      <ul className="flex items-center gap-2">
        <Button
          page={prevPage}
          pathname={pathname}
          search={search}
          type="prev"
          LinkComponent={LinkComponent}
          testId="pagination-prev"
        />
        {totalPages >= 2 && (
          <Page
            page={1}
            currentPage={currentPage}
            pathname={pathname}
            search={search}
            LinkComponent={LinkComponent}
            testId="pagination-firstpage"
          />
        )}
        {showPrevDots && (
          <li className="font-bold" data-testid="pagination-more-prev-dots">
            ...
          </li>
        )}
        {pagesArray.map((page) => {
          return (
            <Page
              key={page}
              page={page}
              currentPage={currentPage}
              pathname={pathname}
              search={search}
              LinkComponent={LinkComponent}
              testId={`pagination-page-${page}`}
            />
          );
        })}
        {showNextDots && (
          <li className="font-bold" data-testid="pagination-more-next-dots">
            ...
          </li>
        )}
        <Page
          page={totalPages}
          currentPage={currentPage}
          pathname={pathname}
          search={search}
          LinkComponent={LinkComponent}
          testId="pagination-lastpage"
        />
        <Button
          page={nextPage}
          pathname={pathname}
          search={search}
          type="next"
          testId="pagination-next"
          LinkComponent={LinkComponent}
        />
      </ul>
    </nav>
  );
};
