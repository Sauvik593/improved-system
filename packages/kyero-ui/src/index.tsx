export * from './Alert';
export * from './ArrowLink';
export * from './Badge';
export * from './Button';
export * from './Checkbox';
export * from './CheckboxList';
export * from './FiltersField';
export * from './Footer';
export * from './LanguageSwitcher';
export * from './Navbar';
export * from './Pagination';
export * from './Radio';
export * from './RadioList';
export * from './SearchInput';
export * from './Select';
export * from './Switch';
export * from './Tooltip';
