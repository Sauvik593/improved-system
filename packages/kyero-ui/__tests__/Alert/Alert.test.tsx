import renderer from 'react-test-renderer';

import { fireEvent, render } from '@testing-library/react';
import { Alert, type AlertProps } from '../../src';
import type { LinkComponentProps } from '../../src/types';

const getProps = (props: Partial<AlertProps> = {}) => ({
  type: 'info',
  fullWidth: false,
  closable: false,
  onClose: undefined,
  children: undefined,
  ...props,
});

const DummyLink = (props: LinkComponentProps) => {
  return (
    <a data-testid="dummy-link" href={props.to as string} className={props.className}>
      {props.children}
    </a>
  );
};

const renderAlert = (props: AlertProps) => (
  <Alert {...props}>
    <p>test</p>
  </Alert>
);

describe('<Alert />', () => {
  it('should render correctly', () => {
    const props = getProps() as AlertProps;
    const alert = renderer.create(renderAlert(props)).toJSON();
    expect(alert).toMatchSnapshot();
  });

  describe('success type', () => {
    it('should render correctly', () => {
      const props = getProps({ type: 'success' }) as AlertProps;
      const alert = renderer.create(renderAlert(props)).toJSON();
      expect(alert).toMatchSnapshot();
    });
  });

  describe('error type', () => {
    it('should render correctly', () => {
      const props = getProps({ type: 'error' }) as AlertProps;
      const alert = renderer.create(renderAlert(props)).toJSON();
      expect(alert).toMatchSnapshot();
    });
  });

  describe('full width', () => {
    it('should render correctly', () => {
      const props = getProps({ fullWidth: true }) as AlertProps;
      const alert = renderer.create(renderAlert(props)).toJSON();
      expect(alert).toMatchSnapshot();
    });
  });

  describe('closable', () => {
    const closeMock = jest.fn();

    it('should render correctly', () => {
      const props = getProps({ closable: true, onClose: closeMock }) as AlertProps;
      const alert = renderer.create(renderAlert(props));
      expect(alert.toJSON()).toMatchSnapshot();
    });

    it('should handle click correctly', () => {
      const props = getProps({ closable: true, onClose: closeMock }) as AlertProps;
      const { getByTestId } = render(renderAlert(props));
      const closeButton = getByTestId('close');

      fireEvent.click(closeButton);

      expect(closeMock).toHaveBeenCalledTimes(1);
    });
  });

  describe('closable with action', () => {
    const closeMock = jest.fn();

    it('should render correctly', () => {
      const props = getProps({
        closable: true,
        onClose: closeMock,
        action: {
          to: '/',
          message: 'Test',
          LinkComponent: DummyLink,
        },
      }) as AlertProps;
      const alert = renderer.create(renderAlert(props));
      expect(alert.toJSON()).toMatchSnapshot();
    });

    it('should have a button', () => {
      const props = getProps({
        closable: true,
        onClose: closeMock,
        action: {
          to: '/testing',
          message: 'Test',
          LinkComponent: DummyLink,
        },
      }) as AlertProps;
      const { getByTestId } = render(renderAlert(props));
      const action = getByTestId('dummy-link');
      expect(action).toBeTruthy();
      expect(action).toHaveAttribute('href', '/testing');
    });
  });
});
