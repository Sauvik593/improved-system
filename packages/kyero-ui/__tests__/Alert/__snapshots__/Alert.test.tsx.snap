// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert /> closable should render correctly 1`] = `
<div
  className="border-1 items-center gap-2 rounded p-3 bg-sunshine-20 border-sunshine-100 inline-flex"
  role="alert"
>
  <div
    className="mr-12 flex items-center gap-2"
  >
    <div
      className="flex"
    >
      <svg
        className="text-sunshine-100"
        fill="none"
        height={20}
        viewBox="0 0 24 24"
        width={20}
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          clipPath="url(#clip0_539_2185)"
        >
          <path
            d="M12 2C10.0222 2 8.08879 2.58649 6.4443 3.6853C4.79981 4.78412 3.51809 6.3459 2.76121 8.17317C2.00433 10.0004 1.8063 12.0111 2.19215 13.9509C2.578 15.8907 3.53041 17.6725 4.92894 19.0711C6.32746 20.4696 8.10929 21.422 10.0491 21.8079C11.9889 22.1937 13.9996 21.9957 15.8268 21.2388C17.6541 20.4819 19.2159 19.2002 20.3147 17.5557C21.4135 15.9112 22 13.9778 22 12C22 9.34784 20.9464 6.8043 19.0711 4.92893C17.1957 3.05357 14.6522 2 12 2ZM10.9286 7C10.9286 6.71584 11.0415 6.44332 11.2424 6.24239C11.4433 6.04145 11.7158 5.92857 12 5.92857C12.2842 5.92857 12.5567 6.04145 12.7576 6.24239C12.9586 6.44332 13.0714 6.71584 13.0714 7V11.2857C13.0714 11.5699 12.9586 11.8424 12.7576 12.0433C12.5567 12.2443 12.2842 12.3571 12 12.3571C11.7158 12.3571 11.4433 12.2443 11.2424 12.0433C11.0415 11.8424 10.9286 11.5699 10.9286 11.2857V7ZM12 17.7143C11.7175 17.7143 11.4413 17.6305 11.2063 17.4735C10.9714 17.3166 10.7883 17.0934 10.6802 16.8324C10.5721 16.5714 10.5438 16.2841 10.5989 16.007C10.654 15.7299 10.7901 15.4754 10.9899 15.2756C11.1896 15.0758 11.4442 14.9397 11.7213 14.8846C11.9984 14.8295 12.2857 14.8578 12.5467 14.9659C12.8077 15.074 13.0308 15.2571 13.1878 15.492C13.3448 15.727 13.4286 16.0032 13.4286 16.2857C13.4286 16.6646 13.2781 17.028 13.0102 17.2959C12.7422 17.5638 12.3789 17.7143 12 17.7143Z"
            fill="currentColor"
          />
        </g>
        <defs>
          <clipPath
            id="clip0_539_2185"
          >
            <rect
              fill="white"
              height="20"
              transform="translate(2 2)"
              width="20"
            />
          </clipPath>
        </defs>
      </svg>
    </div>
    <div
      className="text-sierra-night-100"
    >
      <p>
        test
      </p>
    </div>
  </div>
  <div
    className="ml-auto flex gap-2"
  >
    <button
      className="p-1"
      data-testid="close"
      onClick={[MockFunction]}
    >
      <svg
        className="text-sierra-night-100"
        fill="none"
        viewBox="0 0 24 24"
        width={18}
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          clipPath="url(#clip0_539_1272)"
        >
          <path
            d="M5.4375 18.5619L18.5625 5.43692"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
          />
          <path
            d="M18.5625 18.5619L5.4375 5.43692"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
          />
        </g>
        <defs>
          <clipPath
            id="clip0_539_1272"
          >
            <rect
              fill="currentColor"
              height="24"
              width="24"
            />
          </clipPath>
        </defs>
      </svg>
    </button>
  </div>
</div>
`;

exports[`<Alert /> closable with action should render correctly 1`] = `
<div
  className="border-1 items-center gap-2 rounded p-3 bg-sunshine-20 border-sunshine-100 inline-flex"
  role="alert"
>
  <div
    className="mr-12 flex items-center gap-2"
  >
    <div
      className="flex"
    >
      <svg
        className="text-sunshine-100"
        fill="none"
        height={20}
        viewBox="0 0 24 24"
        width={20}
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          clipPath="url(#clip0_539_2185)"
        >
          <path
            d="M12 2C10.0222 2 8.08879 2.58649 6.4443 3.6853C4.79981 4.78412 3.51809 6.3459 2.76121 8.17317C2.00433 10.0004 1.8063 12.0111 2.19215 13.9509C2.578 15.8907 3.53041 17.6725 4.92894 19.0711C6.32746 20.4696 8.10929 21.422 10.0491 21.8079C11.9889 22.1937 13.9996 21.9957 15.8268 21.2388C17.6541 20.4819 19.2159 19.2002 20.3147 17.5557C21.4135 15.9112 22 13.9778 22 12C22 9.34784 20.9464 6.8043 19.0711 4.92893C17.1957 3.05357 14.6522 2 12 2ZM10.9286 7C10.9286 6.71584 11.0415 6.44332 11.2424 6.24239C11.4433 6.04145 11.7158 5.92857 12 5.92857C12.2842 5.92857 12.5567 6.04145 12.7576 6.24239C12.9586 6.44332 13.0714 6.71584 13.0714 7V11.2857C13.0714 11.5699 12.9586 11.8424 12.7576 12.0433C12.5567 12.2443 12.2842 12.3571 12 12.3571C11.7158 12.3571 11.4433 12.2443 11.2424 12.0433C11.0415 11.8424 10.9286 11.5699 10.9286 11.2857V7ZM12 17.7143C11.7175 17.7143 11.4413 17.6305 11.2063 17.4735C10.9714 17.3166 10.7883 17.0934 10.6802 16.8324C10.5721 16.5714 10.5438 16.2841 10.5989 16.007C10.654 15.7299 10.7901 15.4754 10.9899 15.2756C11.1896 15.0758 11.4442 14.9397 11.7213 14.8846C11.9984 14.8295 12.2857 14.8578 12.5467 14.9659C12.8077 15.074 13.0308 15.2571 13.1878 15.492C13.3448 15.727 13.4286 16.0032 13.4286 16.2857C13.4286 16.6646 13.2781 17.028 13.0102 17.2959C12.7422 17.5638 12.3789 17.7143 12 17.7143Z"
            fill="currentColor"
          />
        </g>
        <defs>
          <clipPath
            id="clip0_539_2185"
          >
            <rect
              fill="white"
              height="20"
              transform="translate(2 2)"
              width="20"
            />
          </clipPath>
        </defs>
      </svg>
    </div>
    <div
      className="text-sierra-night-100"
    >
      <p>
        test
      </p>
    </div>
  </div>
  <div
    className="ml-auto flex gap-2"
  >
    <a
      className="bg-white text-sunshine-100 px-4 py-2 rounded-full text-center border-1 border-sunshine-100"
      data-testid="dummy-link"
      href="/"
    >
      Test
    </a>
    <button
      className="p-1"
      data-testid="close"
      onClick={[MockFunction]}
    >
      <svg
        className="text-sierra-night-100"
        fill="none"
        viewBox="0 0 24 24"
        width={18}
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          clipPath="url(#clip0_539_1272)"
        >
          <path
            d="M5.4375 18.5619L18.5625 5.43692"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
          />
          <path
            d="M18.5625 18.5619L5.4375 5.43692"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
          />
        </g>
        <defs>
          <clipPath
            id="clip0_539_1272"
          >
            <rect
              fill="currentColor"
              height="24"
              width="24"
            />
          </clipPath>
        </defs>
      </svg>
    </button>
  </div>
</div>
`;

exports[`<Alert /> error type should render correctly 1`] = `
<div
  className="border-1 items-center gap-2 rounded p-3 bg-terracotta-10 border-terracotta-100 inline-flex"
  role="alert"
>
  <div
    className="mr-12 flex items-center gap-2"
  >
    <div
      className="flex"
    >
      <svg
        className="text-terracotta-100"
        fill="none"
        height={20}
        viewBox="0 0 24 24"
        width={20}
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          clipPath="url(#clip0_539_2185)"
        >
          <path
            d="M12 2C10.0222 2 8.08879 2.58649 6.4443 3.6853C4.79981 4.78412 3.51809 6.3459 2.76121 8.17317C2.00433 10.0004 1.8063 12.0111 2.19215 13.9509C2.578 15.8907 3.53041 17.6725 4.92894 19.0711C6.32746 20.4696 8.10929 21.422 10.0491 21.8079C11.9889 22.1937 13.9996 21.9957 15.8268 21.2388C17.6541 20.4819 19.2159 19.2002 20.3147 17.5557C21.4135 15.9112 22 13.9778 22 12C22 9.34784 20.9464 6.8043 19.0711 4.92893C17.1957 3.05357 14.6522 2 12 2ZM10.9286 7C10.9286 6.71584 11.0415 6.44332 11.2424 6.24239C11.4433 6.04145 11.7158 5.92857 12 5.92857C12.2842 5.92857 12.5567 6.04145 12.7576 6.24239C12.9586 6.44332 13.0714 6.71584 13.0714 7V11.2857C13.0714 11.5699 12.9586 11.8424 12.7576 12.0433C12.5567 12.2443 12.2842 12.3571 12 12.3571C11.7158 12.3571 11.4433 12.2443 11.2424 12.0433C11.0415 11.8424 10.9286 11.5699 10.9286 11.2857V7ZM12 17.7143C11.7175 17.7143 11.4413 17.6305 11.2063 17.4735C10.9714 17.3166 10.7883 17.0934 10.6802 16.8324C10.5721 16.5714 10.5438 16.2841 10.5989 16.007C10.654 15.7299 10.7901 15.4754 10.9899 15.2756C11.1896 15.0758 11.4442 14.9397 11.7213 14.8846C11.9984 14.8295 12.2857 14.8578 12.5467 14.9659C12.8077 15.074 13.0308 15.2571 13.1878 15.492C13.3448 15.727 13.4286 16.0032 13.4286 16.2857C13.4286 16.6646 13.2781 17.028 13.0102 17.2959C12.7422 17.5638 12.3789 17.7143 12 17.7143Z"
            fill="currentColor"
          />
        </g>
        <defs>
          <clipPath
            id="clip0_539_2185"
          >
            <rect
              fill="white"
              height="20"
              transform="translate(2 2)"
              width="20"
            />
          </clipPath>
        </defs>
      </svg>
    </div>
    <div
      className="text-sierra-night-100"
    >
      <p>
        test
      </p>
    </div>
  </div>
  <div
    className="ml-auto flex gap-2"
  />
</div>
`;

exports[`<Alert /> full width should render correctly 1`] = `
<div
  className="border-1 items-center gap-2 rounded p-3 bg-sunshine-20 border-sunshine-100 flex"
  role="alert"
>
  <div
    className="mr-12 flex items-center gap-2"
  >
    <div
      className="flex"
    >
      <svg
        className="text-sunshine-100"
        fill="none"
        height={20}
        viewBox="0 0 24 24"
        width={20}
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          clipPath="url(#clip0_539_2185)"
        >
          <path
            d="M12 2C10.0222 2 8.08879 2.58649 6.4443 3.6853C4.79981 4.78412 3.51809 6.3459 2.76121 8.17317C2.00433 10.0004 1.8063 12.0111 2.19215 13.9509C2.578 15.8907 3.53041 17.6725 4.92894 19.0711C6.32746 20.4696 8.10929 21.422 10.0491 21.8079C11.9889 22.1937 13.9996 21.9957 15.8268 21.2388C17.6541 20.4819 19.2159 19.2002 20.3147 17.5557C21.4135 15.9112 22 13.9778 22 12C22 9.34784 20.9464 6.8043 19.0711 4.92893C17.1957 3.05357 14.6522 2 12 2ZM10.9286 7C10.9286 6.71584 11.0415 6.44332 11.2424 6.24239C11.4433 6.04145 11.7158 5.92857 12 5.92857C12.2842 5.92857 12.5567 6.04145 12.7576 6.24239C12.9586 6.44332 13.0714 6.71584 13.0714 7V11.2857C13.0714 11.5699 12.9586 11.8424 12.7576 12.0433C12.5567 12.2443 12.2842 12.3571 12 12.3571C11.7158 12.3571 11.4433 12.2443 11.2424 12.0433C11.0415 11.8424 10.9286 11.5699 10.9286 11.2857V7ZM12 17.7143C11.7175 17.7143 11.4413 17.6305 11.2063 17.4735C10.9714 17.3166 10.7883 17.0934 10.6802 16.8324C10.5721 16.5714 10.5438 16.2841 10.5989 16.007C10.654 15.7299 10.7901 15.4754 10.9899 15.2756C11.1896 15.0758 11.4442 14.9397 11.7213 14.8846C11.9984 14.8295 12.2857 14.8578 12.5467 14.9659C12.8077 15.074 13.0308 15.2571 13.1878 15.492C13.3448 15.727 13.4286 16.0032 13.4286 16.2857C13.4286 16.6646 13.2781 17.028 13.0102 17.2959C12.7422 17.5638 12.3789 17.7143 12 17.7143Z"
            fill="currentColor"
          />
        </g>
        <defs>
          <clipPath
            id="clip0_539_2185"
          >
            <rect
              fill="white"
              height="20"
              transform="translate(2 2)"
              width="20"
            />
          </clipPath>
        </defs>
      </svg>
    </div>
    <div
      className="text-sierra-night-100"
    >
      <p>
        test
      </p>
    </div>
  </div>
  <div
    className="ml-auto flex gap-2"
  />
</div>
`;

exports[`<Alert /> should render correctly 1`] = `
<div
  className="border-1 items-center gap-2 rounded p-3 bg-sunshine-20 border-sunshine-100 inline-flex"
  role="alert"
>
  <div
    className="mr-12 flex items-center gap-2"
  >
    <div
      className="flex"
    >
      <svg
        className="text-sunshine-100"
        fill="none"
        height={20}
        viewBox="0 0 24 24"
        width={20}
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          clipPath="url(#clip0_539_2185)"
        >
          <path
            d="M12 2C10.0222 2 8.08879 2.58649 6.4443 3.6853C4.79981 4.78412 3.51809 6.3459 2.76121 8.17317C2.00433 10.0004 1.8063 12.0111 2.19215 13.9509C2.578 15.8907 3.53041 17.6725 4.92894 19.0711C6.32746 20.4696 8.10929 21.422 10.0491 21.8079C11.9889 22.1937 13.9996 21.9957 15.8268 21.2388C17.6541 20.4819 19.2159 19.2002 20.3147 17.5557C21.4135 15.9112 22 13.9778 22 12C22 9.34784 20.9464 6.8043 19.0711 4.92893C17.1957 3.05357 14.6522 2 12 2ZM10.9286 7C10.9286 6.71584 11.0415 6.44332 11.2424 6.24239C11.4433 6.04145 11.7158 5.92857 12 5.92857C12.2842 5.92857 12.5567 6.04145 12.7576 6.24239C12.9586 6.44332 13.0714 6.71584 13.0714 7V11.2857C13.0714 11.5699 12.9586 11.8424 12.7576 12.0433C12.5567 12.2443 12.2842 12.3571 12 12.3571C11.7158 12.3571 11.4433 12.2443 11.2424 12.0433C11.0415 11.8424 10.9286 11.5699 10.9286 11.2857V7ZM12 17.7143C11.7175 17.7143 11.4413 17.6305 11.2063 17.4735C10.9714 17.3166 10.7883 17.0934 10.6802 16.8324C10.5721 16.5714 10.5438 16.2841 10.5989 16.007C10.654 15.7299 10.7901 15.4754 10.9899 15.2756C11.1896 15.0758 11.4442 14.9397 11.7213 14.8846C11.9984 14.8295 12.2857 14.8578 12.5467 14.9659C12.8077 15.074 13.0308 15.2571 13.1878 15.492C13.3448 15.727 13.4286 16.0032 13.4286 16.2857C13.4286 16.6646 13.2781 17.028 13.0102 17.2959C12.7422 17.5638 12.3789 17.7143 12 17.7143Z"
            fill="currentColor"
          />
        </g>
        <defs>
          <clipPath
            id="clip0_539_2185"
          >
            <rect
              fill="white"
              height="20"
              transform="translate(2 2)"
              width="20"
            />
          </clipPath>
        </defs>
      </svg>
    </div>
    <div
      className="text-sierra-night-100"
    >
      <p>
        test
      </p>
    </div>
  </div>
  <div
    className="ml-auto flex gap-2"
  />
</div>
`;

exports[`<Alert /> success type should render correctly 1`] = `
<div
  className="border-1 items-center gap-2 rounded p-3 bg-meadow-100 border-meadow-100 inline-flex"
  role="alert"
>
  <div
    className="mr-12 flex items-center gap-2"
  >
    <div
      className="flex"
    >
      <svg
        className="text-white"
        fill="none"
        viewBox="0 0 24 24"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M12 22C17.5228 22 22 17.5228 22 12C22 6.47715 17.5228 2 12 2C6.47715 2 2 6.47715 2 12C2 17.5228 6.47715 22 12 22ZM16.7071 9.70711C17.0976 9.31658 17.0976 8.68342 16.7071 8.29289C16.3166 7.90237 15.6834 7.90237 15.2929 8.29289L10 13.5858L8.70711 12.2929C8.31658 11.9024 7.68342 11.9024 7.29289 12.2929C6.90237 12.6834 6.90237 13.3166 7.29289 13.7071L9.29289 15.7071C9.68342 16.0976 10.3166 16.0976 10.7071 15.7071L16.7071 9.70711Z"
          fill="currentColor"
          fillRule="evenodd"
        />
      </svg>
    </div>
    <div
      className="text-white"
    >
      <p>
        test
      </p>
    </div>
  </div>
  <div
    className="ml-auto flex gap-2"
  />
</div>
`;
